Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MediaVariant.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataformat.access;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.kitodo.api.dataformat.mets.UseXmlAttributeAccessInterface;
+import org.kitodo.dataformat.metskitodo.MetsType.FileSec.FileGrp;
+
+/**
+ * A variant of the {@code MediaUnit}s for a particular use. By use is meant a
+ * technical form of use. Therefore, the variant has as technical property the
+ * Internet MIME type. Examples of variants include thumbnails, full resolution
+ * images, file ready to print, OCR results, etc. For Production, a variant is
+ * always linked to a MIME type, it is not possible to have mixed MIME types in
+ * the same usage variant.
+ */
+public class MediaVariant implements UseXmlAttributeAccessInterface {
+    /**
+     * Specifying the Internet MIME type for this type of use.
+     */
+    private String mimeType;
+
+    /**
+     * Identifier of use. In terms of METS, this is a string, but if the
+     * generated METS file is to be used with the DFG Viewer this string must be
+     * a prescribed value, depending on its intended use.
+     * 
+     * @see "https://www.zvdd.de/fileadmin/AGSDD-Redaktion/METS_Anwendungsprofil_2.0.pdf#page=12"
+     */
+    private String use;
+
+    /**
+     * Public constructor for a media variant. This constructor can be used with
+     * the service loader to create a new variant.
+     */
+    public MediaVariant() {
+    }
+
+    /**
+     * Constructor that creates a variant from a METS {@code <fileGrp>}.
+     * 
+     * @param fileGrp
+     *            METS {@code <fileGrp>} from which a media variant is to be
+     *            created
+     * @throws IllegalArgumentException
+     *             if the MIME type is mixed within the METS {@code <fileGrp>}
+     */
+    MediaVariant(FileGrp fileGrp) {
+        this.use = fileGrp.getUSE();
+        Set<String> mimeTypes = fileGrp.getFile().parallelStream().map(fileType -> fileType.getMIMETYPE())
```

Comment:
why is getUSE and getMIMETYPE in capital letters?
Sorry, I don't really get what is happening here :(

Reply:
These are the method names that JXC (the generator for JAXB) extracts from METS. In METS, the attributes are written in uppercase letters. I have no influence on that.